cassovary-core/src/main/scala/com/twitter/cassovary/graph/bipartite/BipartiteGraph.scala

```diff
@@ -110,22 +95,52 @@ case class BipartiteSide(nodes: Array[BipartiteNode], numOfNodes: Int, numOfOutE
  * semantic meaning between the left and right sides; but if the graph is of direction
  * BipartiteBoth, then it could carry two semantic
  * meanings in one graph (e.g. list-to-user membership and user-to-list followship).
- *
+ * @param nodeIdsUnique true if node Ids are unique in the whole graph
+ *                      and false if unique only on each side (LHS and RHS)
+ * If the node IDs are unique only on each side, then the Left Nodes IDs are negated,
+ * so that IDs are unique in the whole graph
  */
-class BipartiteGraph(val leftNodes: Array[BipartiteNode], val leftNodeCount: Int,
-                     val leftOutEdgeCount: Long, val rightNodes: Array[BipartiteNode],
+class BipartiteGraph(leftBipartiteNodes: Array[BipartiteNode], val leftNodeCount: Int,
+                     val leftOutEdgeCount: Long, rightBipartiteNodes: Array[BipartiteNode],
                      val rightNodeCount: Int, val rightOutEdgeCount: Long,
-                     val bipartiteGraphDir: BipartiteGraphDir.BipartiteGraphDir) extends Graph[BipartiteNode] {
+                     val bipartiteGraphDir: BipartiteGraphDir.BipartiteGraphDir,
+                     val nodeIdsUnique: Boolean = false) extends Graph[BipartiteNode] {
 
   def this(leftSide: BipartiteSide, rightSide: BipartiteSide,
```

Comment:
Scala allows only one constructor to have default values. So, we need both these constructors so that it doesn't break existing tests. Again, should we remove the default value completely as I said below and change tests?